### ISSUE TITLE
No Redeclaration of Let

### ISSUE THREAD

**Issue by shadowtime2000** [eslint/eslint#13763]
**Please describe what the rule should do:**
It should report an error if someone redeclares a variable made with `let`.

**What new ECMAScript feature does this rule relate to?**
ES6 `const` and `let` variable declarations.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```javascript
let foo = "bar";

var foo = "foobar";
```

```javascript
let foo = "bar";

let foo = "foobar";
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Because this is kind of like `no-const-reassign`.

**Are you willing to submit a pull request to implement this rule?**
Yes.

**Comment by ljharb** [comment]
Both of these are syntax errors, so they shouldn't even parse successfully. Does this code pass eslint currently?

**Comment by shadowtime2000** [comment]
Missed that.
